### ISSUE TITLE
Thread panicked/PoisonError while running local node connected to Nicolas's trusted-peer

### ISSUE THREAD

**Issue by dorin100** [input-output-hk/jormungandr#818]
**Describe the bug**
Thread panicked/PoisonError after running the node for ~3 hours.

**Mandatory Information**
```
jormungandr 0.5.0 (HEAD-98922c77, release, windows [x86_64]) - [rustc 1.36.0 (a53f9df32 2019-07-03)]
jcli 0.5.0 (HEAD-98922c77, release, windows [x86_64]) - [rustc 1.36.0 (a53f9df32 2019-07-03)]
```

**To Reproduce**
1. start local node using `jormungandr ---config node_config.yaml --genesis-block-hash 314c2e572306bc38450b3a4b87a4610bbe4b7ed962b03e05a06a096e6d6e07e7`
2. after ~3 hours I saw the below errors on logs
```
Sep 24 11:29:36.514 INFO ignoring P2P node without an IP address: Node(Node { id: Id([255, 227, 226, 84, 74, 36, 239, 83]), address: None, subscriptions: Subscriptions({}), subscribers: {}, last_gossip: SystemTime { intervals: 132137771977202100 } }), task: network                                                                                                                                                                         thread 'tokio-runtime-worker-1' panicked at 'peer must be present in the map', src\libcore\option.rs:1036:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
thread 'tokio-runtime-worker-2' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', src\libcore\result.rs:999:5
thread 'tokio-runtime-worker-0' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', src\libcore\result.rs:999:5
Sep 24 11:29:39.284thread 'tokio-runtime-worker-1 ' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"INFO', src\libcore\result.rs :999block subscription stream failure: Error { code: Unknown, source: Status { code: Unknown, message: "h2 protocol error: broken pipe" } }:5,                                                                                                                              sub_task: server, task: network                                                                                                                                                                                  Sep 24 11:29:39.294 INFO fragment subscription stream failure: Error { code: Unknown, source: Status { code: Unknown, message: "h2 protocol error: broken pipe" } }, sub_task: server, task: network                     Sep 24 11:29:39.299 INFO gossip subscription stream failure: Error { code: Unknown, source: Status { code: Unknown, message: "h2 protocol error: broken pipe" } }, sub_task: server, task: network                       thread 'tokio-runtime-worker-2' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', src\libcore\result.rs:999:5
thread 'tokio-runtime-worker-3' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', src\libcore\result.rs:999:5
thread 'tokio-runtime-worker-0' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', src\libcore\result.rs:999:5
Sep 24 11:32:40.458 ERRO cannot propagate block to network: send failed because receiver is gone, task: block
Sep 24 11:32:40.502 ERRO cannot propagate block to network: send failed because receiver is gone, task: block
```

**Expected behavior**
No errors/panic messages should be displayed.

**Additional context**
[node_logs.txt](https://github.com/input-output-hk/jormungandr/files/3646236/node_logs.txt)

**Comment by manveru** [comment]
Here another log of this happening
[failure.log](https://github.com/input-output-hk/jormungandr/files/3651600/failure.log)

After the failure, the node seems to be unresponsive, but not crashed, so it wasn't restarted automatically.

**Comment by mzabaluev** [comment]
The root cause is fixed in #811.
